### ISSUE TITLE
Support multiple schemas in diesel.toml

### ISSUE THREAD

**Issue by compressed** [diesel-rs/diesel#1728]
### Versions

- **Diesel:** 1.3
- **Database:** Postgres

### What are you trying to accomplish?

In 1.3, the `diesel.toml` file has been introduced. In the current form, you can only read tables from one schema. I'd like a way to specify multiple schemas that can all be generated into a single `schema.rs` file.

Perhaps you could have multiple `[print-schema]` sections, each one belonging to a different schema, since you may want to apply different filtering depending on the schema. I think all the existing options are good to have available for each schema you want to load from, except probably `patch_file` since it applies to the whole file.

**Comment by sgrif** [comment]
I'm not sure that multiple `print-schema` sections is the way to go, but I think we need to support this somehow.

**Comment by compressed** [comment]
That sounds better to me because, looking again, `patch_file` and `with_docs` are universal options that would apply to all schemas. It would certainly make for a shorter and easier to understand file. I'm not sure if you'd want schema specific `import_types` (I don't use that feature currently).

**Comment by sgrif** [comment]
I don't think we need `import_types` be schema specific. There's no harm in having the imports used by schema 2 in the tables from schema 1 99% of the time. I'm sure someone out there has a very specific case where this doesn't work, but... that's what `patch_file` is for.

**Comment by trsh** [comment]
I like to keep my structure like /models/table1/model.rs, schema.rs, /models/table2/model.rs,schema.rs. So yes I would also appreciate some scenario, where I can split **Large** single schema.rs in to multiple ones.

**Comment by tcmal** [comment]
Something like `[printschema.whatever]` with the usual properties could work if it inherited from the base `printschema`, I can't think of many other ways to do it & support filtering that isn't kinda messy imo.

**Comment by huhlig** [comment]
Something like the following would likely be beneficial

`
# For documentation on how to configure this file,
# see diesel.rs/guides/configuring-diesel-cli

[print_schema]
include_schemas = ["schema1", "schema2"]

[print_schema.schema1]
file = "src/models/schema1.rs"
filter = { except_tables = ["spatial_ref_sys"] }

[print_schema.schema2]
file = "src/models/schema2.rs"
filter = { except_tables = ["spatial_ref_sys"] }
`

**Comment by drusellers** [comment]
I would for sure love multiple schema support. If there is an approved direction I'd be happy to try and take it on.

**Comment by weiznich** [comment]
@drusellers Sorry for the late answer. There is currently no approved direction for how this should be implemented. That does not mean you cannot start to work on this issue, as we are just looking for a reasonable way to configure this. We are open for suggestions here and bringing in a implementation is certainly a argument for a certain way of doing this :wink:

**Comment by drusellers** [comment]
@weiznich totally understand. :) I'll start poking around.

**Comment by jiangxiaoqiang** [comment]
> Something like the following would likely be beneficial
> 
> ```toml
> # For documentation on how to configure this file,
> # see diesel.rs/guides/configuring-diesel-cli
> 
> [print_schema]
> include_schemas = ["schema1", "schema2", "schema3"]
> exclude_schemas = ["pg_catalog", "information_schema"] # Implicit defaults include xor exclude
> patch_file = ""
> 
> [print_schema.schema1]
> file = "src/models/schema1.rs"
> filter = { except_tables = ["spatial_ref_sys"] }
> 
> [print_schema.schema2]
> file = "src/models/schema2.rs"
> filter = { except_tables = ["ignored_system_table"] }
> ```

I tried this way but give tips:


`unknown field `include_schemas`, expected one of `file`, `with_docs`, `filter`, `schema`, `patch_file`, `import_types` for key `print_schema``

**Comment by weiznich** [comment]
@jiangxiaoqiang That's not surprising as this is not implemented yet. Please don't comment on old issues if you don't have anything to add.

**Comment by StuartHadfield** [comment]
Any movement here folks?

**Comment by weiznich** [comment]
@StuartHadfield Please don't comment on old issues if you don't have anything to add.

**Comment by StuartHadfield** [comment]
Hi @weiznich, yeah I am asking if there is any movement on a four year old issue that has been accepted, I'm not commenting for my personal satisfaction.

**Comment by weiznich** [comment]
@StuartHadfield It doesn't matter for me why you comment here, but how. It would be a completely different discussion if you used a different wording in your question, possibly even suggesting that you want to work on this. Again: I consider just asking for updates as rude, as it wastes the time of the maintainer, so stop doing that. Anything that's relevant for this issue is linked above. If there is no entry for some time, that means there is no movement on that topic for this time. This explicitly includes the fact that this topic is waiting for someone to work on.

(And yes, those last two comments are considered off topic here, so they will be marked as off topic.)

**Comment by forest1102** [comment]
I made PR on this issue, #3796. 
Do you have any request on this PR?
https://github.com/diesel-rs/diesel/pull/3796#issue-1899787757
